bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/ProfileCallbackImpl.java

```diff
@@ -64,10 +65,11 @@ public void handleCommand(Command command) {
                 if (ThingHandlerHelper.isHandlerInitialized(thing)) {
                     logger.debug("Delegating command '{}' for item '{}' to handler for channel '{}'", command,
                             link.getItemName(), link.getLinkedUID());
-                    safeCaller.create(handler).onTimeout(() -> {
-                        logger.warn("Handler for thing '{}' takes more than {}ms for handling a command",
-                                handler.getThing().getUID(), SafeCaller.DEFAULT_TIMEOUT);
-                    }).build().handleCommand(link.getLinkedUID(), command);
+                    safeCaller.create(handler).withTimeout(CommunicationManager.THINGHANDLER_EVENT_TIMEOUT)
+                            .onTimeout(() -> {
+                                logger.warn("Handler for thing '{}' takes more than {}ms for handling a command",
+                                        handler.getThing().getUID(), SafeCaller.DEFAULT_TIMEOUT);
```

Comment:
`SafeCaller.DEFAULT_TIMEOUT` -> `CommunicationManager.THINGHANDLER_EVENT_TIMEOUT` so that the logging won't lie